arch/x86/core/swap.S

```diff
@@ -329,6 +329,33 @@ CROHandlingDone:
 
 	movl    %eax, _kernel_offset_to_current(%edi)
 
+#ifdef CONFIG_MEMORY_DOMAIN
+	/* Reset the mem_domain partitions, if configured.*/
```

Comment:
there's no need to do this in assembly. put this logic in _x86_swap_update_page_tables()

Reply:
All these are with the old architecture. I will move these APIs into the correct locations.